.pre-commit-config.yaml

```diff
@@ -0,0 +1,114 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# See https://pre-commit.com for more information
+
+# General excludes, files can also be excluded on a hook level
+exclude: .*\.patch|scripts/tests/.*|velox/external/.*|CMake/third-party/.*
+default_install_hook_types: [pre-commit, pre-push]
+repos:
+  - repo: meta
+    hooks:
+      - id: check-hooks-apply
+      - id: check-useless-excludes
+
+  - repo: https://github.com/pre-commit/pre-commit-hooks
```

Comment:
Nice, these are the minimal ones I would recommend. My other favorite hooks in this collection are:
- `check-executables-have-shebangs`
- `check-shebang-scripts-are-executable`
- `check-yaml` (there are similar validators for JSON and Python, but the YAML one catches lots of CI mishaps)